testplugins/src/main/java/org/spongepowered/test/ScoreTextTest.java

```diff
@@ -0,0 +1,98 @@
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.test;
+
+import org.slf4j.Logger;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.CommandException;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.spec.CommandSpec;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.game.state.GameStartingServerEvent;
+import org.spongepowered.api.plugin.Plugin;
+import org.spongepowered.api.scoreboard.Score;
+import org.spongepowered.api.scoreboard.Scoreboard;
+import org.spongepowered.api.scoreboard.objective.Objective;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.format.TextColors;
+import org.spongepowered.api.text.selector.Selector;
+
+import javax.inject.Inject;
+
+@Plugin(id = "scoretexttest", name = "ScoreText test", description = "A plugin to test projectiles")
+public class ScoreTextTest {
+
+    @Inject
+    private Logger logger;
+
+    @Listener
+    public void onGameStartingServer(GameStartingServerEvent event) {
+        //"/scoreboard objectives add test dummy test"
+        Sponge.getCommandManager().register(
+                this,
+                CommandSpec.builder()
+                        .executor(this::showTest)
+                        .description(Text.of("Creates a dummy objective 'test', adds 1 to it, then shows you the ScoreText"))
+                        .build(),
+                "scoretext"
+        );
+    }
+
+    private CommandResult showTest(CommandSource src, CommandContext ctx) throws CommandException {
+        if (!(src instanceof Player)) {
+            throw new CommandException(Text.of("Must be ran by player"));
```

Comment:
Note that, for a basic command (not a sub command) both throwing a `CommandException` and an empty will end up returning an empty result, the difference is exactly what @ryantheleach said. There is another important difference if an exception is thrown and this is a child command in a `CommandSpec`, it prompts the system to try to run the parent command instead - returning an empty result would not - and this can be a useful guide for whether to throw an exception or not.

So, just as the command elements would throw an exception if the argument string was not in an expected state, the command source would not in an expected state - I'd argue throwing an exception makes sense.